caravel/models.py

```diff
@@ -577,7 +577,19 @@ def query(  # sqla
                 select_exprs += [timestamp_grain]
                 groupby_exprs += [timestamp_grain]
 
-            tf = '%Y-%m-%d %H:%M:%S.%f'
+            # UGLY: I guess correct way is to delegate on SQLAlchemy dialect
```

Comment:
I agree, I got the issue yesterday. I figured out as well it was SQLAlchemy.
And I dont think Its "formating" issue, it's conversion issue. It's converting the datetime in Datetime2 (precision to the nanosecond).
I will try to look into it later.
For now I changed my columns in Datetime2 as a workaround for testing purpose.